ipatests/test_webui/test_pwpolicy.py

```diff
@@ -54,4 +51,214 @@ def test_crud(self):
+        self.facet_button_click('revert')
+
+    @screenshot
+    def test_undo_refresh_revert(self):
+        """
+        Test to verify undo/refresh/revert
+        """
+        self.init_app()
+        self.add_record(pwpolicy.ENTITY, pwpolicy.DATA_RESET)
+        self.navigate_to_record(pwpolicy.PKEY6)
+
+        # test undo
+        self.fill_fields(pwpolicy.DATA_RESET['mod'])
+        mod_field = 0
+        for field in FIELDS:
+            modified_value = (pwpolicy.DATA_RESET['mod'][mod_field][2])
+            self.click_undo_button(field)
+            self.verify_btn_action(field, modified_value)
+            mod_field + 1
+
+        # test refresh
+        mod_field = 0
+        for field in FIELDS:
+            self.fill_fields(pwpolicy.DATA_RESET['mod'])
+            modified_value = (pwpolicy.DATA_RESET['mod'][mod_field][2])
+            self.facet_button_click('refresh')
+            self.verify_btn_action(field, modified_value)
+            mod_field + 1
+
+        # test revert
+        mod_field = 0
+        for field in FIELDS:
+            self.fill_fields(pwpolicy.DATA_RESET['mod'])
+            modified_value = (pwpolicy.DATA_RESET['mod'][mod_field][2])
+            self.facet_button_click('revert')
+            self.verify_btn_action(field, modified_value)
+            mod_field + 1
+
+        # cleanup
+        self.navigate_to_entity(pwpolicy.ENTITY)
+        self.delete_record(pwpolicy.group.PKEY6)
+
+    def verify_btn_action(self, field, mod_value, negative=True):
+        """
+        comparing current value with modified value
+        """
+        current_value = self.get_field_value(field, element="input")
+        if negative:
+            assert current_value != mod_value
+        else:
+            assert current_value == mod_value
+
+    @screenshot
+    def test_verify_measurement_unit(self):
+        """
+        verifying measurement unit for password policy ticket-2437
+        """
+        self.init_app()
+        self.navigate_to_entity(pwpolicy.ENTITY)
+        self.navigate_to_record('global_policy')
+        krbmaxpwdlife = self.get_text('label[name="krbmaxpwdlife"]')
+        krbminpwdlife = self.get_text('label[name="krbminpwdlife"]')
+        krbpwdhistorylen = self.get_text('label[name="krbpwdhistorylength"]')
+        krbpwdfailurecountinterval = \
+            self.get_text('label[name="krbpwdfailurecountinterval"]')
+        krbpwdlockoutduration = \
+            self.get_text('label[name="krbpwdlockoutduration"]')
+        assert "Max lifetime (days)" in krbmaxpwdlife
+        assert "Min lifetime (hours)" in krbminpwdlife
+        assert "History size (number of passwords)" in krbpwdhistorylen
+        assert "Failure reset interval (seconds)" in krbpwdfailurecountinterval
+        assert "Lockout duration (seconds)" in krbpwdlockoutduration
+
+        # cleanup
+        self.delete(group.ENTITY, [group.DATA, group.DATA2, group.DATA3,
```

Comment:
this cleanup, or parts of it, should be in the respected methods so that the tests are self-containing. This way we are relying on specific tests execution order.

Reply:
This cleanup is related to pre_data

Reply:
The issue is that all the delete data is created in `test_misc` but cleaned in `test_verify_measurement_unit`. But those test are not related. Web UI tests don't inherit from IntegrationTest which has `@ordered` decorator to ensure test order. Also Web UI tests don't use this decorator. Thus the reason why it works could be purely random. There can be test execution where `test_verify_measurement_unit` is not the last and `test_misc` is not almost first. I've a feeling that it works only because the tests are sorted alphabetically. 

Each test should prepare it's own data and also clean them.

Reply:
Thank you @pvoborni for comment.
Will fix it.